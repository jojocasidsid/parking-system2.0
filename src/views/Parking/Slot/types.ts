export interface ISlotData {
	id: number
	west: number
	east: number
	south: number
	vehicle: string
	type: 1 | 2 | 3
	parkedType: 0 | 1 | 2 | 3
	parkTime: null | string
	unparkTime: null | string
}

export interface IProps {
	data: ISlotData
	handleLeave: (id: number) => void
	handleCancelReservation: (id: number) => void
}
