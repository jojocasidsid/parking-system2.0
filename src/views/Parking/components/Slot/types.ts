export interface ISlotData {
	id: number
	exitA: number
	exitB: number
	exitC: number
	vehicle: string
	type: 1 | 2 | 3
	parkedType: 0 | 1 | 2 | 3
	parkTime: null | 'string'
	unparkTime: null | 'string'
}

export interface IProps {
	data: ISlotData
	handleLeave: (id: number) => void
}
