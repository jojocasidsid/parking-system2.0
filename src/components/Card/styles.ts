import styled from '@emotion/styled'

export const CardWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-direction: column;
	border-radius: 30px;
	border: 2px solid ${({ theme }) => theme.palette.secondary.main};
`
export const StyledSlotHeader = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`
