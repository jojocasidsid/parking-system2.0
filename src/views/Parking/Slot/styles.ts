import styled from '@emotion/styled'
import DirectionsCarIcon from '@mui/icons-material/DirectionsCar'

export const StyledSlot = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`

export const StyledImage = styled(DirectionsCarIcon)`
	font-size: 45px;
`

export const StyledBody = styled('div')`
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	text-align: center;
`
