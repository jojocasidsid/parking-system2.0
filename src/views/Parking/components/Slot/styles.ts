import styled from '@emotion/styled'
import DirectionsCarIcon from '@mui/icons-material/DirectionsCar'

export const ThumbnailWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-direction: column;
	border-radius: 30px;
	border: 2px solid ${({ theme }) => theme.palette.secondary.main};
`
export const StyledSlot = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`

export const StyledImage = styled(DirectionsCarIcon)`
	font-size: 45px;
`

export const ActionWrapper = styled('div')`
	& > button {
		margin: 20px;
	}
`
