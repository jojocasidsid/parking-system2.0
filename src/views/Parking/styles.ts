import styled from '@emotion/styled'

export const StyledRoot = styled('div')`
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	padding: 20px;
`

export const ParkingWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-wrap: nowrap;
`

export const SlotsWrapper = styled('div')`
	display: grid;
	grid-template-columns: repeat(5, 1fr);
	grid-template-rows: repeat(5, 1fr);
	grid-gap: 10px;
	width: 90vw;
`
