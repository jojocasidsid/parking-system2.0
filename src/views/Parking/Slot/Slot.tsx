import React, { useState, useEffect, useRef } from 'react'

import { Button, Typography } from '@mui/material'
import moment from 'moment'
import pluralize from 'pluralize'

import numbertToType from 'helpers/numberToType/numberToType'

import Card from 'components/Card'

import { getTimeDifference } from 'helpers'
import { IProps } from './types'

import { StyledImage, StyledBody } from './styles'

const Slot = ({ data, handleLeave, handleCancelReservation }: IProps) => {
	const [timeDiff, setTimeDiff] = useState(0)
	const [timeReservation, setTimeReservation] = useState(0)
	const [trigger, setTrigger] = useState(0)
	const counter = useRef(0)

	useEffect(() => {
		const dateNow = moment().format()
		if (data.parkTime) {
			const timeBetween = getTimeDifference(data.parkTime).toFixed(2)
			setTimeDiff(Number(timeBetween))

			const timeUntilReserved = getTimeDifference(dateNow, data.parkTime).toFixed(
				2
			)

			if (Number(timeUntilReserved) > 0) {
				setTimeReservation(Number(timeUntilReserved))
			}
		}

		// trigger useEffect every 20 seconds
		if (counter.current < 20) {
			counter.current += 1
			const timer = setTimeout(() => setTrigger(trigger + 1), 1000)
			return () => clearTimeout(timer)
		}

		return () => {}
	}, [JSON.stringify(data), trigger])

	return (
		<Card
			header={
				<>
					<Typography variant='h1'>{data.id}</Typography>
					<StyledImage color={data.parkedType ? 'primary' : 'secondary'} />
				</>
			}
		>
			<StyledBody>
				{data.parkedType !== 0 ? (
					<Typography variant='h4'>
						Plate: {data.vehicle} || Type: {`${numbertToType(data.parkedType)}`}
						{Boolean(timeReservation) && '|| Reserved'}
					</Typography>
				) : (
					<Typography variant='h3'>Available</Typography>
				)}

				<Typography variant='h3'>
					Parking Type: {numbertToType(data.type)} <br />
				</Typography>
				<br />

				<Typography variant='h4'>
					Distances: [ {pluralize('unit', data.west, true)} ,{' '}
					{pluralize('unit', data.east, true)} ,{' '}
					{pluralize('unit', data.south, true)} ]{' '}
				</Typography>

				<br />
				{data.parkedType ? (
					<Typography variant='h4'>
						{timeReservation
							? pluralize('Hour', timeReservation, true)
							: pluralize('Hour', timeDiff, true)}{' '}
						/ {moment(data.parkTime).format('YYYY-MM-DD HH:mm:ss')}{' '}
					</Typography>
				) : (
					<div />
				)}

				<br />
				{data.parkedType && !data.unparkTime && !timeReservation ? (
					<Button variant='contained' fullWidth onClick={() => handleLeave(data.id)}>
						<Typography variant='h4' color='white'>
							Leave
						</Typography>
					</Button>
				) : (
					<div />
				)}

				{Boolean(timeReservation) && (
					<Button
						variant='contained'
						fullWidth
						color='secondary'
						onClick={() => handleCancelReservation(data.id)}
					>
						<Typography variant='h4' color='white'>
							Cancel Reservation
						</Typography>
					</Button>
				)}

				{data.unparkTime ? (
					<Typography variant='h4'>
						Unpark Time: {moment(data.unparkTime).format('YYYY-MM-DD HH:mm:ss')}{' '}
					</Typography>
				) : (
					<div />
				)}
			</StyledBody>
		</Card>
	)
}

export default Slot
