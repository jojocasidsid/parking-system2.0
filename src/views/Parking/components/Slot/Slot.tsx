import React, { useState, useEffect } from 'react'

import { Button, Typography } from '@mui/material'
import moment from 'moment'
import pluralize from 'pluralize'

import numbertToType from 'helpers/numberToType/numberToType'

import { IProps } from './types'

import { ThumbnailWrapper, StyledImage, StyledSlot } from './styles'

const Slot = ({ data, handleLeave }: IProps) => {
	const [timeDiff, setTimeDiff] = useState(0)

	useEffect(() => {
		const dateNow = moment()
		if (data.parkTime) {
			const dateParked = moment(data.parkTime)
			const timeBetween = moment.duration(dateNow.diff(dateParked))

			setTimeDiff(Math.ceil(timeBetween.asHours()))
		}
	}, [data])

	return (
		<ThumbnailWrapper>
			<StyledSlot>
				<Typography variant='h1'>{data.id}</Typography>

				<StyledImage color={data.parkedType ? 'primary' : 'secondary'} />
			</StyledSlot>

			{data.parkedType ? (
				<Typography variant='h4'>Plate: {data.vehicle}</Typography>
			) : (
				<div />
			)}

			<Typography variant='h3'>
				Slot:{' '}
				{data.parkedType
					? `${numbertToType(data.parkedType)} vehicle is parked`
					: 'Available'}
			</Typography>
			<Typography variant='h3'>
				Parking Type: {numbertToType(data.type)} <br />
			</Typography>
			<br />

			<Typography variant='h4'>
				Distances: [ {pluralize('unit', data.west, true)} ,{' '}
				{pluralize('unit', data.east, true)} , {pluralize('unit', data.south, true)}{' '}
				]{' '}
			</Typography>

			<br />
			{data.parkedType ? (
				<Typography variant='h3'>
					{pluralize('Hour', timeDiff, true)}/
					{moment(data.parkTime).format('YYYY-MM-DD HH:mm:ss')}{' '}
				</Typography>
			) : (
				<div />
			)}

			<br />
			{data.parkedType && !data.unparkTime ? (
				<Button variant='contained' fullWidth onClick={() => handleLeave(data.id)}>
					<Typography variant='h4' color='white'>
						Leave
					</Typography>
				</Button>
			) : (
				<div />
			)}

			{data.unparkTime ? (
				<Typography variant='h3'>
					Unpark Time: {moment(data.unparkTime).format('YYYY-MM-DD HH:mm:ss')}{' '}
				</Typography>
			) : (
				<div />
			)}
		</ThumbnailWrapper>
	)
}

export default Slot
