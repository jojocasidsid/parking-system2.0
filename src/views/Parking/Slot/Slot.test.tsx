import React from 'react'
import renderer from 'react-test-renderer'

// packages
import { ThemeProvider } from '@mui/material/styles'

// config
import theme from 'config/theme'

// component
import Slot from './Slot'

describe('Slot parking Component', () => {
	it('should render slot component without crashing', () => {
		const tree = renderer.create(
			<ThemeProvider theme={theme}>
				<Slot
					data={{
						id: 1,
						west: 1,
						south: 1,
						east: 1,
						vehicle: '123-123',
						type: 1,
						parkedType: 1,
						parkTime: null,
						unparkTime: null,
					}}
					handleLeave={() => {}}
					handleCancelReservation={() => {}}
				/>
			</ThemeProvider>
		)

		expect(tree).toMatchSnapshot()
	})

	it('should render slot component with park time', () => {
		const time: string = '2022-05-06T22:12:32.222Z'
		const tree = renderer.create(
			<ThemeProvider theme={theme}>
				<Slot
					data={{
						id: 1,
						west: 1,
						south: 1,
						east: 1,
						vehicle: '123-123',
						type: 1,
						parkedType: 1,
						parkTime: time,
						unparkTime: null,
					}}
					handleLeave={() => {}}
					handleCancelReservation={() => {}}
				/>
			</ThemeProvider>
		)

		expect(tree).toMatchSnapshot()
	})

	it('should render slot component with later park time', () => {
		const time: string = '2030-05-06T22:12:32.222Z'
		const tree = renderer.create(
			<ThemeProvider theme={theme}>
				<Slot
					data={{
						id: 1,
						west: 1,
						south: 1,
						east: 1,
						vehicle: '123-123',
						type: 1,
						parkedType: 1,
						parkTime: time,
						unparkTime: null,
					}}
					handleLeave={() => {}}
					handleCancelReservation={() => {}}
				/>
			</ThemeProvider>
		)

		expect(tree).toMatchSnapshot()
	})

	it('should render slot component with unpark time', () => {
		const time: string = '2022-05-06T22:12:32.222Z'
		const tree = renderer.create(
			<ThemeProvider theme={theme}>
				<Slot
					data={{
						id: 1,
						west: 1,
						south: 1,
						east: 1,
						vehicle: '123-123',
						type: 1,
						parkedType: 1,
						parkTime: time,
						unparkTime: '2025-05-06T22:12:32.222Z',
					}}
					handleLeave={() => {}}
					handleCancelReservation={() => {}}
				/>
			</ThemeProvider>
		)

		expect(tree).toMatchSnapshot()
	})

	it('should render slot component with available slot', () => {
		const tree = renderer.create(
			<ThemeProvider theme={theme}>
				<Slot
					data={{
						id: 1,
						west: 1,
						south: 1,
						east: 1,
						vehicle: '',
						type: 1,
						parkedType: 0,
						parkTime: '',
						unparkTime: '',
					}}
					handleLeave={() => {}}
					handleCancelReservation={() => {}}
				/>
			</ThemeProvider>
		)

		expect(tree).toMatchSnapshot()
	})
})
