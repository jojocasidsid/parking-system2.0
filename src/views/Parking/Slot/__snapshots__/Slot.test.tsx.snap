// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slot parking Component should render slot component with available slot 1`] = `
<div
  className="css-joqnyj"
>
  <div
    className="css-e98rco"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-17re0f5-MuiTypography-root"
    >
      1
    </h1>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-g9ch1l-MuiSvgIcon-root"
      data-testid="DirectionsCarIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
      />
    </svg>
  </div>
  <div>
    <div
      className="css-j5wv8"
    >
      <h3
        className="MuiTypography-root MuiTypography-h3 css-1dscty8-MuiTypography-root"
      >
        Available
      </h3>
      <h3
        className="MuiTypography-root MuiTypography-h3 css-1dscty8-MuiTypography-root"
      >
        Parking Type: 
        SP
         
        <br />
      </h3>
      <br />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Distances: [ 
        1 unit
         ,
         
        1 unit
         ,
         
        1 unit
         ]
         
      </h4>
      <br />
      <div />
      <br />
      <div />
      <div />
    </div>
  </div>
</div>
`;

exports[`Slot parking Component should render slot component with later park time 1`] = `
<div
  className="css-joqnyj"
>
  <div
    className="css-e98rco"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-17re0f5-MuiTypography-root"
    >
      1
    </h1>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1yauy2g-MuiSvgIcon-root"
      data-testid="DirectionsCarIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
      />
    </svg>
  </div>
  <div>
    <div
      className="css-j5wv8"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Plate: 
        123-123
         || Type: 
        SP
         ||
         
      </h4>
      <h3
        className="MuiTypography-root MuiTypography-h3 css-1dscty8-MuiTypography-root"
      >
        Parking Type: 
        SP
         
        <br />
      </h3>
      <br />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Distances: [ 
        1 unit
         ,
         
        1 unit
         ,
         
        1 unit
         ]
         
      </h4>
      <br />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        0 Hours
         
        / 
        2030-05-07 06:12:32
         
      </h4>
      <br />
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1r01nr4-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-8kdnk8-MuiTypography-root"
        >
          Leave
        </h4>
      </button>
      <div />
    </div>
  </div>
</div>
`;

exports[`Slot parking Component should render slot component with park time 1`] = `
<div
  className="css-joqnyj"
>
  <div
    className="css-e98rco"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-17re0f5-MuiTypography-root"
    >
      1
    </h1>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1yauy2g-MuiSvgIcon-root"
      data-testid="DirectionsCarIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
      />
    </svg>
  </div>
  <div>
    <div
      className="css-j5wv8"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Plate: 
        123-123
         || Type: 
        SP
         ||
         
      </h4>
      <h3
        className="MuiTypography-root MuiTypography-h3 css-1dscty8-MuiTypography-root"
      >
        Parking Type: 
        SP
         
        <br />
      </h3>
      <br />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Distances: [ 
        1 unit
         ,
         
        1 unit
         ,
         
        1 unit
         ]
         
      </h4>
      <br />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        35.94 Hours
         
        / 
        2022-05-07 06:12:32
         
      </h4>
      <br />
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1r01nr4-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-8kdnk8-MuiTypography-root"
        >
          Leave
        </h4>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div />
    </div>
  </div>
</div>
`;

exports[`Slot parking Component should render slot component with unpark time 1`] = `
<div
  className="css-joqnyj"
>
  <div
    className="css-e98rco"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-17re0f5-MuiTypography-root"
    >
      1
    </h1>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1yauy2g-MuiSvgIcon-root"
      data-testid="DirectionsCarIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
      />
    </svg>
  </div>
  <div>
    <div
      className="css-j5wv8"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Plate: 
        123-123
         || Type: 
        SP
         ||
         
      </h4>
      <h3
        className="MuiTypography-root MuiTypography-h3 css-1dscty8-MuiTypography-root"
      >
        Parking Type: 
        SP
         
        <br />
      </h3>
      <br />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Distances: [ 
        1 unit
         ,
         
        1 unit
         ,
         
        1 unit
         ]
         
      </h4>
      <br />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        35.94 Hours
         
        / 
        2022-05-07 06:12:32
         
      </h4>
      <br />
      <div />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Unpark Time: 
        2025-05-07 06:12:32
         
      </h4>
    </div>
  </div>
</div>
`;

exports[`Slot parking Component should render slot component without crashing 1`] = `
<div
  className="css-joqnyj"
>
  <div
    className="css-e98rco"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-17re0f5-MuiTypography-root"
    >
      1
    </h1>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1yauy2g-MuiSvgIcon-root"
      data-testid="DirectionsCarIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
      />
    </svg>
  </div>
  <div>
    <div
      className="css-j5wv8"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Plate: 
        123-123
         || Type: 
        SP
         ||
         
      </h4>
      <h3
        className="MuiTypography-root MuiTypography-h3 css-1dscty8-MuiTypography-root"
      >
        Parking Type: 
        SP
         
        <br />
      </h3>
      <br />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        Distances: [ 
        1 unit
         ,
         
        1 unit
         ,
         
        1 unit
         ]
         
      </h4>
      <br />
      <h4
        className="MuiTypography-root MuiTypography-h4 css-rprzex-MuiTypography-root"
      >
        0 Hours
         
        / 
        Invalid date
         
      </h4>
      <br />
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1r01nr4-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-8kdnk8-MuiTypography-root"
        >
          Leave
        </h4>
      </button>
      <div />
    </div>
  </div>
</div>
`;
